File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var searchRange = function(nums, target) {
    let num1 = nums.indexOf(target)
    let num2 = nums.lastIndexOf(target)
    let resutl = [-1,-1]
    if(num1 === -1){
        return resutl
    }else {
        return [num1,num2]
    }

};

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cs
public class Solution {
    public int[] SearchRange(int[] nums, int target) {


        int leftindex =returnindex(nums, target, true);
        int rightindex = returnindex(nums, target,false );
        return new int[]  {leftindex,rightindex};
    }

        public int returnindex(int[] nums, int target, bool istrue){
            int low=0,high=nums.Length-1; int result = -1 ;

            while(low <= high){
                int middle = (low + high)/2;
                if(nums[middle] == target){
                    result = middle;
                    if(istrue){
                        high = middle - 1;
                    }else{
                        low = middle + 1;
                    }
                }
                else if(nums[middle] > target){
                    high = middle -1;
                }else{
                    low = middle + 1;
                }
            }

            return result ;
        }
        
    }
